src/main/java/com/redhat/devtools/lsp4ij/features/codeBlockProvider/LSPCodeBlockProvider.java

```diff
@@ -0,0 +1,201 @@
+                                              @Nullable Character openBraceChar,
+                                              int openBraceOffset,
+                                              @Nullable Character closeBraceChar,
+                                              int closeBraceOffset) {
+        Document document = editor.getDocument();
+        List<SelectionRange> selectionRanges = LSPSelectionRangeSupport.getSelectionRanges(file, document, offset);
+        if (!ContainerUtil.isEmpty(selectionRanges)) {
+            // Convert the selection ranges into text ranges
+            Set<TextRange> textRanges = new LinkedHashSet<>(selectionRanges.size());
+            for (SelectionRange selectionRange : selectionRanges) {
+                textRanges.add(LSPSelectionRangeSupport.getTextRange(selectionRange, document));
+                for (SelectionRange parentSelectionRange = selectionRange.getParent();
+                     parentSelectionRange != null;
+                     parentSelectionRange = parentSelectionRange.getParent()) {
+                    textRanges.add(LSPSelectionRangeSupport.getTextRange(parentSelectionRange, document));
+                }
+            }
+
+            // Find containing text ranges that are bounded by brace pairs
+            CharSequence documentChars = document.getCharsSequence();
+            List<TextRange> containingTextRanges = new ArrayList<>(textRanges.size());
+            for (TextRange textRange : textRanges) {
+                if (textRange.getLength() > 1) {
+                    int startOffset = textRange.getStartOffset();
+                    int endOffset = textRange.getEndOffset();
+
+                    char startChar = documentChars.charAt(startOffset);
+                    char endChar = documentChars.charAt(endOffset - 1);
+
+                    // If aligned on an open brace and this ends with the expected close brace, use it
+                    if ((startOffset == openBraceOffset)) {
+                        if ((closeBraceChar != null) && (closeBraceChar == endChar)) {
+                            return textRange;
+                        } else {
+                            return null;
+                        }
+                    }
+                    // If aligned on a close brace and this starts with the expected open brace, use it
+                    else if (((endOffset - 1) == closeBraceOffset)) {
+                        if ((openBraceChar != null) && (openBraceChar == startChar)) {
+                            return textRange;
+                        } else {
+                            return null;
+                        }
+                    }
+                    // Otherwise see if it starts and ends with a known brace pair and we'll find the "closest" below
+                    else if ((openBraceOffset == -1) && (closeBraceOffset == -1) && LSPCodeBlockUtils.isCodeBlockStartChar(file, startChar)) {
```

Comment:
Otherwise we see whether or not the selection range starts and ends with a supported brace pair and, if so, we add it to the list of containing code blocks.